### PR TITLE
DEMO-37: Add travis badge on readme so status is obivious for everyone + fix travis

### PR DESCRIPTION
**Pull request by nicolas-bastien**
https://jira.ez.no/browse/DEMO-37

+
https://jira.ez.no/browse/EZP-26729

Todo:
- [ ] Simplify diff and move to ezplatform repo _(this PR is doing to much)_

### CODE REVIEW

**Comment by nicolas-bastien** [comment]
@clash82 @bdunogier @andrerom 

I open this PR cause I think with have to discuss about the subject.

Right now demo build is the same as the ezplatform one with demo installation.

I think we should simplify the build to focus on the demo code and give developers an understandable way to build their project ?

From my point of view demo can : 

- composer install
- run demo installer
- run checkstyle on /src

then if we have more features : 
- implement test and run them

So this will validate the install process of the project.

I really want to have your feedback before fixing this one please.

**Comment by andrerom** [comment]
Hi @nicolas-bastien,

A bit unsure what you propose here, something you already discussed with @bdunogier ?

**Comment by nicolas-bastien** [comment]
Hi @andrerom, yes it's a thing we agree on with @bdunogier.

So I will chan travis.yml to only install the application as there is no test right now.

This repo miss ez robot parsing for checkstyle, can you add it ?
(Would be great if you can explain me how it works too)

**Comment by nicolas-bastien** [comment]
Hi @andrerom, this is rebased

File: .travis.yml [hunk 1]
```diff
@@ -35,27 +20,16 @@ branches:
...
-after_failure:
-  # Will show us the full log of container's main processes (not counting shell process above running php and behat)
-  - docker-compose logs -t --tail="all"
-  # Will show us what is up, and how long it's been up
-  - docker ps -s
+  - phpenv config-rm xdebug.ini
+  - echo "memory_limit=-1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
+  - travis_retry composer selfupdate
+  - composer config -g github-oauth.github.com "d0285ed5c8644f30547572ead2ed897431c1fc09"
+  - cp app/config/parameters.yml.dist app/config/parameters.yml
```
Comment (bdunogier) [inline .travis.yml]: Why is this necessary ? composer install should take care of it.

**Review by bdunogier — COMMENTED** [review]
Looks okay to me.

What we discussed with Nicolas, André, is that running the kernel / platform test suites on the demo was pointless. What we could run is actual tests (like behat) of the demo itself, but not the core ones.

Do you agree with the principle ?

**Comment by andrerom** [comment]
not sure I'm a fan, might hide issue, and it diverges from ezplatform on how testing is done completely, making merges always conflict in the future.

**Comment by nicolas-bastien** [comment]
@andrerom I understand the merge thing but this bring the question : 'Is merging ezplatform to an existing project is the good solution to upgrade it ?"

I guess ezplatform demo code should participate to show ez good practice to customers and I'm not sure that we suggest to our customer to test their project this way ?

**Comment by andrerom** [comment]
travis file is for us, it's more important that we make sure to catch issues then for it to be example for others. We can place it in gitignore files to make sure it is stripped form archives for instance.

**Comment by bdunogier** [comment]
> 'Is merging ezplatform to an existing project is the good solution to upgrade it ?"

It is the solution we have for the time being, and the one we apply to the demo, in order to apply our best practices.

**Comment by bdunogier** [comment]
@andrerom I don't really see the added value of running the core tests on the demo platform. They haven't caught anything useful so far, that I know of at least.

One option could be to remove `.travis.yml` from the tags. Are they needed here ?

**Comment by nicolas-bastien** [comment]
@bdunogier @andrerom ok this is a big topic, my comment was a little fast ;-)

So I think we can do something simple to remove merge confilct on travis.yml file by using .gitattributes

A detail example can be found here : https://gist.github.com/tmaybe/4c9d94712711229cd506

and the summury is : -

- add a .gitattributes file to the demon with : ".travis.yml  merge=ours"
- before merging run the command : git config merge.ours.driver true

Do you think it's reasonable here ?
